bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/scheduler2/DelegatedScheduler.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.internal.scheduler2;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.smarthome.core.scheduler2.CronJob;
+import org.eclipse.smarthome.core.scheduler2.Scheduler;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ *
+ * @author Peter Kriens - initial contribution and API
+ *
+ */
+@Component(servicefactory = true)
+public class DelegatedScheduler implements Scheduler {
+
+    private InternalSchedulerImpl delegate;
+    private final Set<Closeable> closeables = new HashSet<>();
+
+    @Deactivate
+    void close() {
+        while (true) {
+            Closeable c;
+            synchronized (closeables) {
+                if (closeables.isEmpty()) {
+                    return;
+                }
+                Iterator<Closeable> iterator = closeables.iterator();
+                c = iterator.next();
+                iterator.remove();
+            }
+
+            try {
+                c.close();
+            } catch (Throwable t) {
```

Comment:
Do not catch `Throwable`'s you will also catch for example `OutOfMemoryError`s, this will not improve the stabilitly and predictability of the code.